sipyco/ssl_tools.py

```diff
@@ -0,0 +1,30 @@
+import ssl
+import logging
+
+logging.basicConfig(level=logging.DEBUG)
+logger = logging.getLogger(__name__)
+
+def create_ssl_context(local_cert=None, local_key=None, peer_cert=None, server_mode=False):
+    """Create an SSL context with mutual authentication.
+
+    :param local_cert: Certificate file. Providing this enables SSL.
+    :param local_key: Private key file. Required when local_cert is provided.
+    :param peer_cert: Peer's certificate file to trust. Required when local_cert is provided.
+    :param server_mode: If True, create a server context, otherwise client context.
+    """
+    if local_cert is None:
+        return None
```

Comment:
not a fan of delegating this to this function; why not check if local_cert is None in the other places instead? Can't ``create_ssl_context`` as the function implies if there's no cert provided after all